include/picongpu/plugins/adios/ADIOSWriter.hpp

```diff
@@ -539,11 +539,13 @@ class ADIOSWriter : public IIOBackend
             if (nComponents > 1)
                 datasetName << "/" << name_lookup_tpl[c];
 
+            std::string dataset = datasetName.str( );
```

Comment:
Good catch!

As a minor naming consideration. It is currently weird that the new variable for data set name is called `dataset` and the stream used to get this name called `datasetName`. E.g. with `recordName` in the same function it is more natural.

Maybe do a little refactoring with separating [this logic](https://github.com/ComputationalRadiationPhysics/picongpu/blob/f4672c5d84502fecd9a0f653c7b436ddc0df6e22/include/picongpu/plugins/adios/ADIOSWriter.hpp#L537-L540) into a new function, or at least renaming like `datasetName` -> `datasetNameStream` and `dataset` -> `datasetName`?

Reply:
Add-up: maybe a new function is even not needed, and smth like that that would suffice (did not check, but to give an idea):
```cpp
std::string datasetName = recordName;
if (nComponents > 1)
    datasetName += "/" + name_lookup_tpl[c];
```